test/jdk/java/util/Base64/TestEncodingDecodingLength.java

```diff
@@ -21,55 +21,91 @@
+// the public API would require which has shown to cause intermittent issues
 public class TestEncodingDecodingLength {
 
-    public static void main(String[] args) {
-        int size = Integer.MAX_VALUE - 8;
-        byte[] inputBytes = new byte[size];
-        byte[] outputBytes = new byte[size];
+    // A value large enough to test the desired memory conditions in encode and decode
+    private static final int LARGE_MEM_SIZE = Integer.MAX_VALUE - 8;
+    private static final Base64.Decoder DECODER = Base64.getDecoder();
+    private static final Base64.Encoder ENCODER = Base64.getEncoder();
 
-        // Check encoder with large array length
-        Base64.Encoder encoder = Base64.getEncoder();
-        checkOOM("encode(byte[])", () -> encoder.encode(inputBytes));
-        checkIAE("encode(byte[] byte[])", () -> encoder.encode(inputBytes, outputBytes));
-        checkOOM("encodeToString(byte[])", () -> encoder.encodeToString(inputBytes));
-        checkOOM("encode(ByteBuffer)", () -> encoder.encode(ByteBuffer.wrap(inputBytes)));
-
-        // Check decoder with large array length,
-        // should not throw any exception
-        Arrays.fill(inputBytes, (byte) 86);
-        Base64.Decoder decoder = Base64.getDecoder();
-        decoder.decode(inputBytes);
-        decoder.decode(inputBytes, outputBytes);
-        decoder.decode(ByteBuffer.wrap(inputBytes));
+    // Effectively tests that encode(byte[] src, byte[] dst) throws an
+    // IllegalArgumentException with array sized near Integer.MAX_VALUE. All the
+    // encode() methods call encodedOutLength(), which is where the OOME is expected
+    @Test
+    public void largeEncodeIAETest() throws IllegalAccessException,
+            InvocationTargetException, NoSuchMethodException {
+        Method m = getMethod(ENCODER,
+                "encodedOutLength", int.class, boolean.class);
+        // When throwOOME param is false, encodedOutLength should return -1 in
+        // this situation, which encode() uses to throw IAE
+        assertEquals(-1, m.invoke(ENCODER, LARGE_MEM_SIZE, false));
     }
 
-    private static final void checkOOM(String methodName, Runnable r) {
+    // Effectively tests that the overloaded encode() and encodeToString() methods
+    // throw OutOfMemoryError with array/buffer sized near Integer.MAX_VALUE
+    @Test
+    public void largeEncodeOOMETest() throws IllegalAccessException, NoSuchMethodException {
+        Method m = getMethod(ENCODER,
+                "encodedOutLength", int.class, boolean.class);
         try {
-            r.run();
-            throw new RuntimeException("OutOfMemoryError should have been thrown by: " + methodName);
-        } catch (OutOfMemoryError er) {}
+            m.invoke(ENCODER, LARGE_MEM_SIZE, true);
+        } catch (InvocationTargetException ex) {
+            Throwable rootEx = ex.getCause();
+            assertEquals(OutOfMemoryError.class, rootEx.getClass(), "00ME should be thrown");
```

Comment:
Sorry if it is intentional, but I wonder if you meant "OOME" instead of "00ME" here?

Reply:
Good catch, you sometimes see what you want to see

Reply:
That was unexpected, wonder how I did that unconsciously...